src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy

```diff
@@ -156,6 +156,43 @@ class FileEventFunctionsStressTest extends AbstractFileEventFunctionsTest {
         watcher.close()
     }
 
+    @Requires({ !Platform.current().linux })
+    def "can stop watching a deep hierarchy when it has been deleted"() {
+        given:
+        def watchedDirectoryDepth = 10
+
+        def watchedDir = new File(rootDir, "watchedDir")
+        assert watchedDir.mkdir()
+        List<File> previousRoots = [watchedDir]
+        watchedDirectoryDepth.times { depth ->
+            previousRoots = previousRoots.collectMany { root ->
+                createSubdirs(root, 2)
+            }
+        }
+
+        def callback = new FileWatcherCallback() {
+            @Override
+            void pathChanged(FileWatcherCallback.Type type, String path) {
+                assert !path.empty
+            }
+
+            @Override
+            void reportError(Throwable ex) {
+                ex.printStackTrace()
+                uncaughtFailureOnThread << ex
+            }
+        }
+
+        when:
+        def watcher = startNewWatcher(callback)
+        watcher.startWatching(watchedDir)
+        Thread.sleep(500)
```

Comment:
Can we do better than sleeping here? If we want all the directories to be deleted, then we should have a latch doing that. If we want to stop in the middle of the process, we should have a latch for that. WDYT? Can we maybe do better than sleeping in other cases maybe?

Reply:
Even if we leave the sleep here, there should be a comment explaining what we are waiting for. Same for other places where we sleep in stress tests.

Reply:
This sleep was for making sure no changes from creating the directories come in before we start deleting them. I replaced it by `waitForChangeEventLatency()`.